pkg/sql/catalog/tabledesc/validate.go

```diff
@@ -0,0 +1,1207 @@
+// Copyright 2021 The Cockroach Authors.
```

Comment:
Note: here lives the table validation logic now. Unlike for type descriptors the table validation logic itself didn't need a lot of changes so I sincerely don't think this should impede review.